docs/reference/technology/connectivity/test-lora.md

```diff
@@ -0,0 +1,289 @@
+
+## LoRaWAN Device Classes
+
+LoRaWAN specifications define three distinct types of Device Classes.
+
+* Class A
+* Class B
+* Class C
+
+A LoRaWAN Device always starts as a Class A Device. It can later switch to another class if required. However, Class B and Class C are mutually exclusive.  
+
+### Class A
+
+Class A is a mandatory Device Class. All LoRaWAN devices must implement a Class A. This class includes battery powered sensors/actuators with no latency constraints. This is the most energy efficient communication Class. 
+
+In Class A, a communication cycle is always initiated by the Device. When a Device transmits a datagram, it opens two receive wndows after specific delays. Timings of these delays and the lengths of receive windows themselves are subjected to regional constraints. The transmission on the other hand is need based. However, it is scheduled or transmitted based upon the duty cycle restrictions following an Aloha like mechanism.  
+
+<span class="images">![](https://s3-us-west-2.amazonaws.com/mbed-os-docs-images/class_a_timing.png)<span>Figure 2: Class A timing diagram</span></span>
+
+### Class B
+
+Class B devices allow for receive slots at scheduled times. For this purpose, the Base Station sends out a time synchronous beacon. This makes sure that at this particular time, the receiver, i.e., the Device is listening. The Class B devices are suitable for latency constrained devices and the devices use slotted communication synchronized with the network beacon. 
+
+The Base Stations transmit a beacon every 128 seconds and all Class B nodes are assigned a time slot within the 128 second cycle and are told when to listen. You can, for instance, tell a node to listen every tenth time slot, and when this comes up, it allows for a downlink message to be transmitted. However this example is just for the purpose of understanding, in actuality, the slots are systematically randomized every beacon period.  
+
+<span class="images">![](https://s3-us-west-2.amazonaws.com/mbed-os-docs-images/class_b_timing.png)<span>Figure 3: Class B timing diagram</span></span>
+
+Each beacon is preceded by Beacon Guard time and no ping slot can be placed in that time period. A ping slot is a 30ms long unit which can be assigned to a Class B device. Beacon Reserved is a time period when the actual beacon is sent. Beacon Window is the time period when ping slots can be opened and assigned. 
+
+* Beacon Period = 128 seconds
+* Beacon Reserved = 2.120 seconds
+* Beacon Guard = 3 seconds
+* Beacon Window = 122.880 seconds
+
+In total there can be 4096, 30ms long ping slots in the Beacon Window. 
+
+**NOTE: Currently Mbed LoRaWAN Stack doesn't support Class B.**
+
+### Class C
+
+Class C devices are supposed to be main powered or that kind of devices who have sufficient amount of power supply available. These devices tend to remain in listening mode always when they are not transmitting. 
```

Comment:
Or when you want to do multicast, f.e. firmware updates.

Reply:
It's a general statement that covers both.